libraries/chain/db_market.cpp

```diff
@@ -929,7 +929,7 @@ bool database::fill_settle_order( const force_settlement_object& settle, const a
 { try {
    bool filled = false;
 
-   auto issuer_fees = pay_market_fees( account_object(), get(receives.asset_id), receives);
+   auto issuer_fees = pay_market_fees( nullptr, get(receives.asset_id), receives);
```

Comment:
I still didn't get why settle orders don't share market fee. IMHO it's a bug. Need a hardfork to fix it though. Will create a new issue.